app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+require 'saml/settings'
+
+class AuthController < ApplicationController
+  skip_before_action :authenticate_user!
+  skip_before_action :verify_authenticity_token
+
+  def new
+    request = OneLogin::RubySaml::Authrequest.new
+    redirect_to(request.create(saml_settings))
+  end
+
+  def callback
+    response = OneLogin::RubySaml::Response.new(params[:SAMLResponse],
+                                                settings: saml_settings)
+    if response.is_valid?
+      user = User.from_saml_callback(response.attributes)
+      if user
+        sign_in_and_redirect user
+      else
+        flash.alert = 'User does not have a valid GIDS system account. ' \
+                      'Contact the system admin for further assistance.'
+        redirect_to root_url + '?auth=fail'
+      end
+    else
+      Rails.logger.info("Invalid SAML response. Errors: #{response.errors.join("\n")}")
```

Comment:
is there the potential for any PII to be included in the response errors?

Reply:
Not as per the [source code](https://github.com/onelogin/ruby-saml/blob/51ebceee5ef6e529159f88e18bcbd3f3ae6c53b5/lib/onelogin/ruby-saml/response.rb#L355).
I will go ahead and remove this line to be completely sure.